app/main.py

```diff
@@ -1 +1,21 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import (NotWearingMaskError,
+                        NotVaccinatedError,
+                        OutdatedVaccineError)
+
+
+def go_to_cafe(friends, cafe) -> str:
+    total = 0
+
+    for i in range(len(friends)):
+        try:
+            cafe.visit_cafe(friends[i])
+        except (NotVaccinatedError, OutdatedVaccineError):
+            return 'All friends should be vaccinated'
+        except NotWearingMaskError:
+            total += 1
+
```

Comment:
1. The current implementation returns a message immediately upon the first vaccination failure. This behavior is correct if you want to complete the check immediately. But if you want to check all your friends, you can change the logic.
2. Break complex designs into individual steps. This will make the code easier to understand.
3. Make sure messages are clear and leave no ambiguity.